sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -270,6 +273,146 @@ class Analyzer(
   }
 
   /**
+   * Rewrite the [[Exists]] [[In]] with left semi join or anti join.
+   */
+  object RewriteFilterSubQuery extends Rule[LogicalPlan] with PredicateHelper {
+    def unapply(condition: Expression): Option[(Expression, Seq[Expression])] = {
+      if (condition.resolved == false) {
+        return None
+      }
+
+      val conjuctions = splitConjunctivePredicates(condition).map(_ transformDown {
+          // Remove the Cast expression for SubQueryExpression.
+          case Cast(f: SubQueryExpression, BooleanType) => f
+        }
+      )
+
+      val (subqueries, others) = conjuctions.partition(c => c.isInstanceOf[SubQueryExpression])
```

Comment:
This is not supported, as the code right below, it will throws exception if this happens.